packages/eslint-plugin/src/util/misc.ts

```diff
@@ -242,20 +242,19 @@ type NodeWithKey =
 function getStaticMemberAccessValue(
   node: NodeWithKey,
   { sourceCode }: RuleContext<string, unknown[]>,
-): string | undefined {
+): string | null {
   const key =
     node.type === AST_NODE_TYPES.MemberExpression ? node.property : node.key;
   if (!node.computed) {
     return key.type === AST_NODE_TYPES.Literal
       ? `${key.value}`
       : (key as TSESTree.Identifier | TSESTree.PrivateIdentifier).name;
   }
-  const value = getStaticValue(key, sourceCode.getScope(node))?.value as
```

Comment:
using optional chaining was not correct. we need to distinguish between a return value of `null`, which means "no static value", versus a return value of `{ value: null }`, which means "access the property called `'null'`.